sdks/python/gen_protos.py

```diff
@@ -311,12 +311,6 @@ def ensure_grpcio_exists():
   try:
     from grpc_tools import protoc  # pylint: disable=unused-import
   except ImportError:
-    if platform.system() == 'Windows':
```

Comment:
Tox is following the build-system https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#build-system-requirement as per PEP 517. Since setup.py doesn't follow/have `build-system` where we could install build requirements to build sdist package, we need to install the build-requirements.txt during runtime. so we need to do that for windows, macOS and ubuntu.